packages/wast-parser/src/index.js

```diff
@@ -1,13 +1,14 @@
 // @flow
 
 import * as parser from "./grammar";
+import validate from "./validator";
 const { tokenize } = require("./tokenizer");
 
 export function parse(source: string): Program {
   const tokens = tokenize(source);
 
   // We pass the source here to show code frames
   const ast = parser.parse(tokens, source);
-
+  validate(ast);
```

Comment:
Actually, I don't know what's the best place for this check. I used to run the validations here but it annoyed me when I wanted to just test the parser. I moved it right in the module "compilation".

What do you think? It's not a parse error but a validation one.

Reply:
Yeah, that’s a good point, having validation in multiple places will confused things. I initially added it here to mirror the functionality of wat2wasm,but that doesn’t mean it’s right for this project. I’ll move it :-)

Reply:
ok thanks